spec/acceptance/reboot_immediate_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper_acceptance'
+
+describe 'Reboot Module - Reboot Immediately and Explicit Immediately' do
+
+  def apply_reboot_manifest(agent, reboot_manifest)
+    apply_manifest_on agent, reboot_manifest
+    retry_shutdown_abort(agent)
+  end
+
+  reboot_manifest = <<-MANIFEST
+notify { 'step_1':
+}
+~>
+reboot { 'now':
+}
+  MANIFEST
+
+  reboot_immediate_manifest = <<-MANIFEST
+notify { 'step_1':
+}
+~>
+reboot { 'now':
+  apply => immediately
+}
+  MANIFEST
+
+  context 'windows agent' do
+    windows_agents.each do |agent|
+      it 'Reboot Immediately with Refresh' do
+        apply_reboot_manifest(agent, reboot_manifest)
+      end
+      it 'Reboot Immediately explicit with Refresh' do
```

Comment:
whenever you use "with" in a description string, this indicates that you have a hidden context here. Consider this structure:

```
describe 'immediate rebooting' do
  context 'when refreshing implicitly' do
    let(:manifest) do .. end
    agents.each do |agent|
      it 'reboots the node' { ... }
    end
  end
  context 'when refreshing explicitly' do
    let(:manifest) do .. end
    agents.each do |agent|
      it 'reboots the node' { ... }
    end
  end
end
```

For further DRY-ness, you could move the `retry_shutdown_abort(agent)` to a cleanup hook, see https://relishapp.com/rspec/rspec-core/v/3-7/docs/hooks/before-and-after-hooks , or https://github.com/puppetlabs/pdk/blob/960c0caf17607243e251827a3438292bb3863a72/spec/unit/pdk/generate/module_spec.rb#L12-L15 for an example.